content/script/code.js

```diff
@@ -110,4 +109,22 @@ function timeInDialect() {
 
     document.getElementById('bernese').innerHTML = "Es isch "+minute+before+hour;
 
+}
+
+/* Update the theme, call with arguments 'themename' or 'standard' to reset theme. */
+// Note: theme is reset at page-reload because it is not yet saved in local/session storage or a cookie.
+
+function changeTheme(theme) {
+    var themeList = ['theme1', 'theme2', 'theme3'];
+    for (style in themeList) {
+        document.querySelector('html').classList.remove(themeList[style]);
+    }
+    if (theme != 'standard') {
+        document.querySelector('html').classList.add(theme);
+    }
+
+    // This part updates the weatherwidget attributes and reinitializes the widget.
+    document.querySelector('a').setAttribute('data-basecolor', getComputedStyle(document.documentElement).getPropertyValue('--color1'));
+    document.querySelector('a').setAttribute('data-textcolor', getComputedStyle(document.documentElement).getPropertyValue('--color0'));
+    __weatherwidget_init()

```

Comment:
This function can initialize the widget. We change the color properties of the widget using the theme colors and then reinitialize it.
It is hacky but it works.

Reply:
Why do you use querySelectors? Wouldn't selecting by id be more accurate and future-proof?